polynote-server/src/main/scala/polynote/server/NotebookManager.scala

```diff
@@ -0,0 +1,65 @@
+package polynote.server
+
+import java.io.File
+import java.util.concurrent.ConcurrentHashMap
+
+import cats.Monad
+import cats.effect.{ContextShift, Fiber, IO}
+import cats.effect.concurrent.Semaphore
+import polynote.server.repository.NotebookRepository
+
+/**
+  * This is how the sessions interact with notebooks. The sessions don't think about writing changes etc. They'll
+  * open a [[SharedNotebook]] and from there get a [[NotebookRef]] in exchange for an output queue. Their interactions
+  * with the notebook will go through the NotebookRef.
+  */
+abstract class NotebookManager[F[_]](implicit F: Monad[F]) {
+
+  def getNotebook(path: String): F[SharedNotebook[F]]
+
+  def listNotebooks(): F[List[String]]
```

Comment:
Given that `Notebook` is also a type, maybe we can change this to something like `listNotebookPaths`? I'd expect `listNotebooks()` to look something like this:

```scala
def listNotebooks: F[List[Notebook]] = listNotebookPaths.flatMap(getNotebook)
```

Reply:
Sure, that's a good point.